### PR TITLE
MediaSDK video backend: Windows support

### PR DESCRIPTION
**Pull request by mshabunin**
### This pullrequest changes

Adds support for Windows version of MediaSDK. Specific `DeviceHandler` instance is empty, internal MediaSDK DirectX device will be used. 

Several warnings have been fixed.

### CODE REVIEW

File: modules/videoio/src/cap_mfx_common.hpp [hunk 1]
```diff
@@ -302,7 +303,25 @@ class VAHandle : public DeviceHandler {
     int file;
 };
 
-// TODO: Windows specific
+#endif
+
+// Windows specific
+#ifdef _MSC_VER
```
Comment (alalek) [inline modules/videoio/src/cap_mfx_common.hpp]: The same problem: detection of platform via compiler is wrong.
`_WIN32` should be here.

File: modules/videoio/src/cap_mfx_common.hpp [hunk 2]
```diff
@@ -302,7 +303,25 @@ class VAHandle : public DeviceHandler {
...
+
+#include <Windows.h>
+inline void sleep(unsigned long msec) { Sleep(msec); }
```
Comment (alalek) [inline modules/videoio/src/cap_mfx_common.hpp]: It is not "msec" in Unix: https://linux.die.net/man/3/sleep

**Comment by mshabunin** [comment]
Fixed.
Thanks for comments.

**Comment by vpisarev** [comment]
looks good to me! :+1:

**Pull request merged**